java/java-ranger-regression/WBS_prop1/Main.java

```diff
@@ -0,0 +1,305 @@
+			   }  else { 
+				   if ((WBS_Node_WBS_BSCU_Switch2 == 3)) {
+				      Nor_Pressure = ((WBS_Node_WBS_SelectorValve_Switch / 4) * 3);
+				   } else { 
+					   if ((WBS_Node_WBS_BSCU_Switch2 == 4)) { 
+					      Nor_Pressure = WBS_Node_WBS_SelectorValve_Switch;
+					   } else { 
+					      Nor_Pressure = 0;
+					   }
+				   }
+			   }
+		   }
+	   }
+
+	//	Debug.printPC("PC before(WBS_Node_WBS_BSCU_Command_PedalCommand_Switch1 == 0) [region 4] = ");
+
+	   if ((WBS_Node_WBS_BSCU_Command_PedalCommand_Switch1 == 0)) {
+	      Alt_Pressure = 0; 
+	   }  else {
+		   if ((WBS_Node_WBS_BSCU_Command_PedalCommand_Switch1 == 1)) {
+		      Alt_Pressure = (WBS_Node_WBS_AS_MeterValve_Switch / 4); 
+		   }  else { 
+			   if ((WBS_Node_WBS_BSCU_Command_PedalCommand_Switch1 == 2)) {
+			      Alt_Pressure = (WBS_Node_WBS_AS_MeterValve_Switch / 2);
+			   } else { 
+				   if ((WBS_Node_WBS_BSCU_Command_PedalCommand_Switch1 == 3)) {
+				      Alt_Pressure = ((WBS_Node_WBS_AS_MeterValve_Switch / 4) * 3);
+				   } else { 
+					   if ((WBS_Node_WBS_BSCU_Command_PedalCommand_Switch1 == 4)) {
+					      Alt_Pressure = WBS_Node_WBS_AS_MeterValve_Switch; 
+					   } else { 
+					      Alt_Pressure = 0;
+					   }
+				   }
+			   }
+		   }
+	   }
+	//	Debug.printPC("PC @fter(WBS_Node_WBS_BSCU_Command_PedalCommand_Switch1 == 0) [Region 4] = ");
+
+
+	   WBS_Node_WBS_rlt_PRE2 = Nor_Pressure; 
+
+	   WBS_Node_WBS_BSCU_rlt_PRE1 = WBS_Node_WBS_BSCU_Switch2; 
+
+	   WBS_Node_WBS_BSCU_SystemModeSelCmd_rlt_PRE = Sys_Mode;
+//	   WBS_Node_WBS_BSCU_SystemModeSelCmd_rlt_PRE += WBS_Node_WBS_BSCU_SystemModeSelCmd_Logical_Operator6 ? 1 : 0;
+
+		// Assertions added by MWW: these are the truth values for "my" version of the model - may not be true for this code, but should be
+		// consistent between SPF and Veritest-SPF.
+
+		// I prefer ite to !a || b for implications.
+
+		// This assertion should prove:
+        //boolean myassert = (PedalPos > 0 && PedalPos <= 4 && !Skid) ? (Alt_Pressure > 0 || Nor_Pressure > 0) : true;
+        //assert(myassert);
+		//assert((PedalPos > 0 && PedalPos <= 4 && !Skid) ? (Alt_Pressure > 0 || Nor_Pressure > 0) : true);
+
+		// This assertion should fail:
+		 assert((PedalPos > 0 && PedalPos <= 4 && !Skid) ? (Alt_Pressure > 0) : true);
+
+		// This assertion may fail (depending on encoding):
+		//Debug.printPC("PC before assertion ((PedalPos > 0 && PedalPos <= 4 && !Skid) ? (Nor_Pressure > 0) : true) = ");
+
+		// assert((PedalPos > 0 && PedalPos <= 4 && !Skid) ? (Nor_Pressure > 0) : true);
+
+		// This assertion should fail:
+		 //assert((PedalPos > 0 && PedalPos <= 4) ? (Alt_Pressure > 0 || Nor_Pressure > 0) : true);
+
+		// This assertion should also fail:
+		//assert((PedalPos > 0 && !Skid) ? (Alt_Pressure > 0 || Nor_Pressure > 0) : true);
+	}
+
+
+	public static void main(String[] args) {
+		Main Main = new Main();
+		Main.update(Verifier.nondetInt(), Verifier.nondetBoolean(), Verifier.nondetBoolean());
```

Comment:
Is there a specific reason, why this function is called 10 times and this is not done in a for loop?
Might you add a comment?

Otherwise, this seems to be code supposed to run in a while(true)-loop. If that is the case, I would advocate for a second version of this test case with a while-loop.

Reply:
There's no specific reason for not using a loop. I've fixed this with a loop in this test case itself.